src/modules/accounting/staking_vaults.py

```diff
@@ -0,0 +1,244 @@
+from typing import List, cast, Any
+
+from eth_typing import ChecksumAddress
+from oz_merkle_tree import StandardMerkleTree
+from web3 import Web3
+from web3.module import Module
+
+from src.modules.accounting.types import VaultData, VaultsData, VaultsMap, VaultTreeNode
+from src.providers.consensus.client import ConsensusClient
+from src.providers.consensus.types import Validator
+from src.providers.execution.contracts.staking_vault import StakingVaultContract
+from src.providers.execution.contracts.vault_hub import VaultHubContract
+from src.providers.ipfs import CID, MultiIPFSProvider
+from src.types import BlockStamp
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class VaultProof:
+    id: int
+    valuationWei: int
+    inOutDelta: int
+    fee: int
+    sharesMinted: int
+    leaf: str
+    proof: List[str]
+
+
+VaultToValidators = dict[ChecksumAddress, list[Validator]]
+
+
+class StakingVaults(Module):
+    w3: Web3
+    ipfs_client: MultiIPFSProvider
+    cl: ConsensusClient
+    vault_hub: VaultHubContract
+
+    def __init__(self, w3: Web3, cl: ConsensusClient, ipfs: MultiIPFSProvider, vault_hub: VaultHubContract) -> None:
+        super().__init__(w3)
+
+        self.w3 = w3
+        self.ipfs_client = ipfs
+        self.cl = cl
+        self.vault_hub = vault_hub
+
+    def _load_vault(self, address: ChecksumAddress) -> StakingVaultContract:
+        return cast(
+            StakingVaultContract,
+            self.w3.eth.contract(
+                address=address,
+                ContractFactoryClass=StakingVaultContract,
+                decode_tuples=True,
+            ),
+        )
+
+    def get_vaults(self, blockstamp: BlockStamp) -> VaultsMap:
+        vault_count = self.vault_hub.get_vaults_count(blockstamp)
+        pending_deposits = self.cl.get_pending_deposits(blockstamp)
+        deposit_map = dict[str, int]()
+
+        for deposit in pending_deposits:
+            if deposit.withdrawal_credentials not in deposit_map:
+                deposit_map[deposit.withdrawal_credentials] = 0
+
+            deposit_map[deposit.withdrawal_credentials] += Web3.to_wei(int(deposit.amount), 'gwei')
+
+        vaults = VaultsMap()
+        for vault_ind in range(vault_count):
+            vault_socket = self.vault_hub.vault_socket(vault_ind, blockstamp)
+
+            balance_wei = self.w3.eth.get_balance(
+                self.w3.to_checksum_address(vault_socket.vault), block_identifier=blockstamp.block_hash
+            )
+
+            vault = self._load_vault(vault_socket.vault)
+            vault_in_out_delta = vault.in_out_delta(blockstamp)
+            print(vault_in_out_delta)
+            vault_withdrawal_credentials = vault.withdrawal_credentials(blockstamp)
+            print(vault_withdrawal_credentials)
+
+            pending_deposit = 0
+            if vault_withdrawal_credentials in deposit_map:
+                pending_deposit = deposit_map[vault_withdrawal_credentials]
```

Comment:
Hmmm, looks like a bug, is there 
```suggestion
                pending_deposit += deposit_map[vault_withdrawal_credentials]
```
supposed to be?

Reply:
If yes, there is test for this one need to add

Reply:
changed